server/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java

```diff
@@ -350,11 +343,14 @@ public void decrypt(char[] password) throws GeneralSecurityException, IOExceptio
             int numEntries = input.readInt();
             while (numEntries-- > 0) {
                 String setting = input.readUTF();
-                EntryType entryType = EntryType.valueOf(input.readUTF());
+                if (formatVersion <= 3) {
```

Comment:
technically this should only be valid for formatVersion 3 since previous versions were serialized differently

Reply:
I pushed a commit for this.